---
title: "explore ccs results"
author: "Tyler Starr"
date: "1/5/2023"
output:
  github_document:
    toc: true
    html_preview: false
editor_options: 
  chunk_output_type: inline
---
This notebook analyzes sites whose mutation effects deviate most strongly among the variant RBD backgrounds.

```{r setup, message=FALSE, warning=FALSE, error=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = T)
knitr::opts_chunk$set(dev.args = list(png = list(type = "cairo")))

#list of packages to install/load
packages = c("yaml","data.table","tidyverse","gridExtra","bio3d")
#install any packages not already installed
installed_packages <- packages %in% rownames(installed.packages())
if(any(installed_packages == F)){
  install.packages(packages[!installed_packages])
}
#load packages
invisible(lapply(packages, library, character.only=T))

#read in config file
config <- read_yaml("config.yaml")


```
Session info for reproducing environment:
```{r print_sessionInfo}
sessionInfo()
```


## Setup

Read in tables of ccs reads processed by alignparse

```{r input_data}
dt <- data.table(read.csv(file="results/process_ccs/MERS_aligned.csv",stringsAsFactors=F))
#dt <- data.table(read.csv(file="results/process_ccs/PDF2180_aligned.csv",stringsAsFactors=F))
```


## Look at split reads

Do things look any different for the positions where we did in-house mutagenesis?

```{r query}
#how many fwd-split with mutations at the MERS 31 in-house mutated position (codon TGC positions 91-93)
dt[query_name %like% "fwd" & gene_mutations %like% "T91"] #1836 rows, of the ten mutants I see, five have errant mutations (distant indels and one mut)
dt[gene_mutations %like% "T91"] #5382 total rows with T91, therefore ~34%
dt[query_name %like% "fwd" & gene_mutations %like% "G92"] #2001 rows, of the ten mutants I see, five have errant mutations (distant indels and one mut)

#how many fwd-split with mutants at say position 88
dt[query_name %like% "fwd" & gene_mutations %like% "A88"] #4330 rows, of the ten mutants I see, five have errant mutations (distant indels and one mut)
dt[gene_mutations %like% "A88"] #10460 total rows with these mutants, therefore ~41%

#how many fwd-split with mutants at say position 880
dt[query_name %like% "fwd" & gene_mutations %like% "T80"] #3545 rows, of the ten mutants I see, five have errant mutations (distant indels and one mut)
dt[gene_mutations %like% "T80"] #11377 total rows with this mutation, therefore 31%

#what about MERS in-house position 64 (nt 190-192 codon AGT)
dt[query_name %like% "fwd" & gene_mutations %like% "A190"] #1573 rows, of the ten mutants I see, 3 have errant mutations (distant indels and one mut)
dt[gene_mutations %like% "A190"] #4711 total rows with thiese mutatns

#compare to 300
dt[query_name %like% "fwd" & gene_mutations %like% "A300"] #3749 rows, of the ten mutants I see, 6 have errant mutations (distant indels and one mut)
dt[gene_mutations %like% "A300"] #10305 total of these mutants


```

AACGGTGACGTGGCCG

with mers data open, look at the 51-5 clone dimer
```{r look at 53-7}
dt[barcode_sequence %like% "AACGGTGACGTGGCCG"]



```

With PDF-2180 data, looking at the 53-7 clone dimer

```{r look at 53-7}
dt[barcode_sequence %like% "CCGGAAAGACTTT"]

dt[gene_mutations %like% "C590A T591C"]


```
